core/io/resource_loader.cpp

```diff
@@ -613,6 +624,21 @@ Ref<Resource> ResourceLoader::load_threaded_get(const String &p_path, Error *r_e
 			}
 			return Ref<Resource>();
 		}
+
+		// Support userland requesting on the main thread before the load is reported to be complete.
+		if (Thread::is_main_thread() && !load_token->local_path.is_empty()) {
+			const ThreadLoadTask &load_task = thread_load_tasks[load_token->local_path];
+			while (load_task.status == THREAD_LOAD_IN_PROGRESS) {
+				if (!_ensure_load_progress()) {
+					// This local poll loop is not needed.
+					break;
+				}
+				thread_load_lock.~MutexLock();
```

Comment:
This makes me think a branch I made some time ago to allow unlocking and re-locking of `MutexLock` is needed, it was just a hypothetical idea then but will revive it and open a PR if it works well still, to avoid potentially hacky solutions

Reply:
Good point, but I think this may not be really a good thing to do depending on the kind of mutex now we have #93701. I haven't checked it yet, but I'm suspicious of this.

Reply:
The issue was something else in the end.